cmd/oras/internal/option/packer.go

```diff
@@ -97,20 +97,19 @@ func decodeJSON(filename string, v interface{}) error {
 	return json.NewDecoder(file).Decode(v)
 }
 
-// parseAnnotationFlags resharps annotationslice to k-v type and updates annotations
-func parseAnnotationFlags(ManifestAnnotations []string, annotations map[string]map[string]string) error {
+// parseAnnotationFlags parses annotation flags into a map.
+func parseAnnotationFlags(flags []string, annotations map[string]map[string]string) error {
 	manifestAnnotations := make(map[string]string)
-	for _, anno := range ManifestAnnotations {
-		parts := strings.SplitN(anno, "=", 2)
-		if len(parts) != 2 {
-			return errAnnotationFormat
+	for _, anno := range flags {
+		key, val, success := strings.Cut(anno, "=")
+		if !success {
+			return fmt.Errorf("%w: %s", errAnnotationFormat, anno)
 		}
-		parts[0], parts[1] = strings.TrimSpace(parts[0]), strings.TrimSpace(parts[1])
-		if _, ok := manifestAnnotations[parts[0]]; ok {
-
-			return fmt.Errorf("found annotation key, %v, more than once, %w", parts[0], errAnnotationDuplication)
+		key, val = strings.TrimSpace(key), strings.TrimSpace(val)
```

Comment:
Is there reason to trim spaces? Trim space for `key` seems reasonable but what if spaces in `val` are intended?

For example, `--annotation "greet=$(echo hello world)"` in bash.

Reply:
Well Noted